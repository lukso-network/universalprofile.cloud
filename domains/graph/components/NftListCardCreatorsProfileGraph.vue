<script setup lang="ts">
type Props = {
  creator?: Creator
  hasName?: boolean
  count?: number
}

const props = defineProps<Props>()
const profileAvatar = useProfileAvatar(props.creator, 24)
</script>

<template>
  <div class="flex h-6">
    <div
      v-if="count"
      class="paragraph-inter-10-semi-bold flex size-6 items-center justify-center rounded-full bg-neutral-100 outline outline-2 outline-neutral-90"
    >
      +{{ count }}
    </div>
    <lukso-profile
      v-else
      size="x-small"
      :profile-url="profileAvatar?.url"
      :profile-address="creator?.address"
      has-identicon
    ></lukso-profile>
    <div class="pl-1" v-if="hasName">
      <div class="paragraph-inter-10-semi-bold text-neutral-60">
        {{ $formatMessage('asset_created_by') }}
      </div>
      <lukso-username
        :name="creator?.name"
        :address="creator?.address"
        size="x-small"
        class="flex"
        name-color="neutral-20"
      ></lukso-username>
    </div>
  </div>
</template>
