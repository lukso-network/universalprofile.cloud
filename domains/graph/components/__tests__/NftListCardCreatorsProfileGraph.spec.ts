import { renderSuspended } from '@nuxt/test-utils/runtime'
import { beforeEach, describe, expect, it } from 'vitest'

import NftListCardCreatorsProfileGraph from '../NftListCardCreatorsProfileGraph.vue'

describe('NftListCardCreatorsProfileGraph', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('render loading state', async () => {
    const component = await renderSuspended(NftListCardCreatorsProfileGraph)

    expect(component.html()).toMatchSnapshot()
  })

  it('render empty creator', async () => {
    const component = await renderSuspended(NftListCardCreatorsProfileGraph, {
      props: {
        creator: {} as Creator,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('render creator', async () => {
    const component = await renderSuspended(NftListCardCreatorsProfileGraph, {
      props: {
        creator: {
          address: '0x123',
          name: 'John Doe',
          profileImage: [
            {
              url: 'https://example.com/profile-image.jpg',
            },
          ],
        } as Creator,
        hasName: true,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('render count', async () => {
    const component = await renderSuspended(NftListCardCreatorsProfileGraph, {
      props: {
        creator: {
          address: '0x123',
          name: 'John Doe',
          profileImage: [
            {
              url: 'https://example.com/profile-image.jpg',
            },
          ],
        } as Creator,
        count: 1,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  describe('when small size', () => {
    it('render loading state', async () => {
      const component = await renderSuspended(NftListCardCreatorsProfileGraph, {
        props: {
          isSmall: true,
        },
      })

      expect(component.html()).toMatchSnapshot()
    })

    it('render creator', async () => {
      const component = await renderSuspended(NftListCardCreatorsProfileGraph, {
        props: {
          creator: {
            address: '0x123',
            name: 'John Doe',
            profileImage: [
              {
                url: 'https://example.com/profile-image.jpg',
              },
            ],
          } as Creator,
          hasName: true,
          isSmall: true,
        },
      })

      expect(component.html()).toMatchSnapshot()
    })

    it('render count', async () => {
      const component = await renderSuspended(NftListCardCreatorsProfileGraph, {
        props: {
          creator: {
            address: '0x123',
            name: 'John Doe',
            profileImage: [
              {
                url: 'https://example.com/profile-image.jpg',
              },
            ],
          } as Creator,
          count: 1,
          isSmall: true,
        },
      })

      expect(component.html()).toMatchSnapshot()
    })
  })
})
