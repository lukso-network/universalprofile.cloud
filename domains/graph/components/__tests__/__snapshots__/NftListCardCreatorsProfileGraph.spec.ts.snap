// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NftListCardCreatorsProfileGraph > render count 1`] = `
"<div id="test-wrapper">
  <div class="flex h-10">
    <div class="flex items-center justify-center rounded-full bg-neutral-100 outline outline-2 outline-neutral-90 paragraph-inter-14-semi-bold size-10"> +1</div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`NftListCardCreatorsProfileGraph > render creator 1`] = `
"<div id="test-wrapper">
  <div class="flex h-10">
    <lukso-profile size="small" profile-url="https://example.com/profile-image.jpg" profile-address="0x123" has-identicon=""></lukso-profile>
    <div class="flex flex-col justify-center gap-1 pl-2">
      <div class="paragraph-inter-10-semi-bold text-neutral-60">Created by</div>
      <lukso-username name="John Doe" address="0x123" size="x-small" class="flex" name-color="neutral-20"></lukso-username>
    </div>
  </div>
</div>"
`;

exports[`NftListCardCreatorsProfileGraph > render empty creator 1`] = `
"<div id="test-wrapper">
  <div class="flex h-10">
    <lukso-profile size="small" has-identicon=""></lukso-profile>
    <!--v-if-->
  </div>
</div>"
`;

exports[`NftListCardCreatorsProfileGraph > render loading state 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;

exports[`NftListCardCreatorsProfileGraph > when small size > render count 1`] = `
"<div id="test-wrapper">
  <div class="flex h-6">
    <div class="flex items-center justify-center rounded-full bg-neutral-100 outline outline-2 outline-neutral-90 paragraph-inter-10-semi-bold size-6"> +1</div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`NftListCardCreatorsProfileGraph > when small size > render creator 1`] = `
"<div id="test-wrapper">
  <div class="flex h-6">
    <lukso-profile size="x-small" profile-url="https://example.com/profile-image.jpg" profile-address="0x123" has-identicon=""></lukso-profile>
    <div class="flex flex-col justify-center pl-1">
      <div class="paragraph-inter-10-semi-bold text-neutral-60">Created by</div>
      <lukso-username name="John Doe" address="0x123" size="x-small" class="flex" name-color="neutral-20"></lukso-username>
    </div>
  </div>
</div>"
`;

exports[`NftListCardCreatorsProfileGraph > when small size > render loading state 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;
